### PR TITLE
Fix: Error msg should return first non-empty msg

### PR DESCRIPTION
**Pull request by cloudxxx8**
##  PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

errors.CommonEdgeX.Message() might return an empty string if it is just a wrapper.

Issue Number: Fix https://github.com/edgexfoundry/go-mod-core-contracts/issues/290

## What is the new behavior?
errors.CommonEdgeX.Message() should return the first non-empty message.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: errors/types.go [hunk 1]
```diff
@@ -100,6 +100,14 @@ func (ce CommonEdgeX) DebugMessages() string {
 
 // Message returns the first level error message without further details.
 func (ce CommonEdgeX) Message() string {
+	if ce.message == "" && ce.err != nil {
+		if w, ok := ce.err.(CommonEdgeX); ok {
+			return w.message
```
Comment (lenny-intel) [inline errors/types.go:105]: Shouldn't this be `w.Message()` so it is recursive.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: errors/types.go (hunk 1 above)
Comment (cloudxxx8) [reply errors/types.go:105]: thanks, @lenny-intel 
my bad, and I added one more unit test to verify the recursive

**Comment by cloudxxx8** [comment]
rebased

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**